Assets/Mirror/Runtime/NetworkManager.cs

```diff
@@ -748,7 +775,33 @@ public static void Shutdown()
 
         #region Server Internal Message Handlers
 
-        void OnServerConnectInternal(NetworkConnection conn, ConnectMessage connectMsg)
+        void OnServerAuthenticateInternal(NetworkConnection conn, ConnectMessage connectMsg)
+        {
+            if (LogFilter.Debug) Debug.Log("NetworkManager.OnServerAuthenticateInternal");
+
+            if (authenticator != null)
+            {
+                // if we have an authenticator then let it handle authentication
+                authenticator.OnServerAuthenticateInternal(conn);
+            }
+            else
+            {
+                // otherwise authenticate immediately
+                OnServerAuthenticated(conn);
+            }
+        }
+
+        // called after successful authentication
+        void OnServerAuthenticated(NetworkConnection conn)
+        {
+            // set connection to authenticated
+            conn.isAuthenticated = true;
+
+            // proceed with the login handshake by calling OnServerConnect
+            OnServerConnectInternal(conn);
+        }
+
+        void OnServerConnectInternal(NetworkConnection conn)
```

Comment:
Move this code inside OnServerAuthenticated

Reply:
Can't - it'll break things